src/App.js

```diff
@@ -1,37 +1,54 @@
 import React from 'react';
-import { Router, Route } from 'react-router-dom';
+import { Router, Route, Switch } from 'react-router-dom';
 import { connect } from 'react-redux';
+
 import { PrivateRoute } from './PrivateRoute.js';
+
 import { history } from './helpers';
 import { alertActions } from './actions';
+
 import { HomePage } from './components/HomePage';
-import { LoginPage } from './components/LoginPage';
-import { RegisterPage } from './components/RegisterPage';
+import LoginPage from './components/LoginPage';
+import RegisterPage from './components/RegisterPage';
+
+import AlertBanner from './components/AlertBanner';
 
 export class App extends React.Component {
     constructor(props) {
         super(props);
 
-        const { dispatch } = this.props;
+        const { alertClear } = this.props;
         history.listen((location, action) => {
+            const keepAlert = location.pathname === '/login' &&
+                !!location.state && location.state.status === 'Registration successful';
+            if (!keepAlert) {
+                alertClear();
+            }
```

Comment:
I'm not confident about my usage of the listen function here. This is my first time using the session history manager, and having the listener set up here in the given code led me to believe it was for clearing alerts upon navigation. Without the extra test, calling `alertClear()` also removed the registration confirmation, although come to think of it, I did dispatch the success alert prior to calling `history.push()`.

Either way, the App test file isn't happy about the `alertClear` prop when not importing the connected component.

Reply:
Removed the need for a conditional by redirecting before the alert is dispatched when registration succeeds: 587af75